e2e/test/iothub/messaging/MessageReceiveE2ETests.cs

```diff
@@ -536,47 +536,55 @@ await Task
             await serviceClient.CloseAsync().ConfigureAwait(false);
         }
 
-        private async Task ReceiveMessagesSentBeforeSubscriptionAsync(TestDeviceType type, Client.TransportType transport)
+        private async Task ReceiveMessagesSentBeforeSubscriptionAsync(TestDeviceType type, Client.TransportType transportType)
         {
             using TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, s_devicePrefix, type).ConfigureAwait(false);
-            DeviceClient deviceClient = testDevice.CreateDeviceClient(transport);
-            var testDeviceCallbackHandler = new TestDeviceCallbackHandler(deviceClient, testDevice, Logger);
-
-            using var serviceClient = ServiceClient.CreateFromConnectionString(TestConfiguration.IoTHub.ConnectionString);
-
-            (Message msg, string payload, string p1Value) = ComposeC2dTestMessage(Logger);
-
-            // Subscribe to receive C2D messages over the callback.
-            await testDeviceCallbackHandler.SetMessageReceiveCallbackHandlerAsync().ConfigureAwait(false);
-
-            // Now dispose and reinitialize the client instance.
-            deviceClient.Dispose();
-            deviceClient = null;
+            using DeviceClient deviceClient1 = testDevice.CreateDeviceClient(transportType);
```

Comment:
what is `deviceClient1` doing here? It appears that we just open and close it without any other actions.

Reply:
The comment below explains it. With MQTT, if a new device hasn't connected yet, then the service client can't send a C2D message to it. I've found this via trial and error.